packages/bierzo-wallet/src/logic/csvBuilder/index.ts

```diff
@@ -0,0 +1,52 @@
+export interface CsvRow {
+  readonly id: string;
+  readonly recepient: string;
+  readonly sender: string;
+  readonly quantity: string;
+  readonly fractionalDigits: string;
+  readonly tokenTicker: string;
+  readonly feeQuantity: string;
+  readonly feeFractionalDigits: string;
+  readonly feeTokenTicker: string;
+  readonly time: string;
+  readonly note: string;
+  [key: string]: string;
```

Comment:
Can you remove this signature, since it is not very type safe?

Reply:
If I will remove it, I will get error from TS because in this case it can not access property by name.

Reply:
But do you need this index signature when you do the following instead?

```ts
const parsedRow: string[] = [
  row.id,
  row.recepient,
  row.sender,
  row.quantity,
  row.fractionalDigits,
  row.tokenTicker,
  row.feeQuantity,
  row.feeFractionalDigits,
  row.feeTokenTicker,
  row.time,
  row.note,
].map(content => `"${content}"`);
```

Reply:
In this case if we will update CsvRow interface then we will need to update this map values too. Easily to miss.

Reply:
Right, but only 2 places need to be in sync and it makes everything else much easier.

With `this._fields = Object.keys(header);` you have no guarantee that `header` contains only `CsvRow` fields. It can have more, unsupported fields.

So we get
1. Guaranteed content
2. Explicit order
3. Removal of usafe index signature
4. No `_fields` necessary
5. `parse` can become static